### DISCUSSION TITLE
How to pass multiple values in a key in params for ERCOT Public API GET request?

### DISCUSSION THREAD

**Discussion by MandyZhang010** [ercot/api-specs#38] [Q&A] [answered]
Hi,

I am trying to use Python to pull market prices via the public API. The data I am looking at is the Real-Time SPP, which has a parameter 'settlementPoint', and I would like to pass a few location there. I've tried params={'settlementPoint':['Point_A', 'Point_B']} and params={'settlementPoint':'Point_A,Point_B'}, but none of them seems to work.

Would anyone please advise? Thank you!

**Comment by cimview** [comment]
The API spec implies that this parameter is a simple string. so you'd probably have to make one API call for each settlement point.  

I've noticed the latency is really high for each of these API calls though, so you might be better off not specifying the settlement point to get all of them, and then filter the results.

**Reply by MandyZhang010** [reply]
Thanks for the comment!

Agree 100% on the latency. There are 900+ settlement points in ERCOT and it's a painful long wait to get all of them, not to mention all those request failures it may cause. I will do a separate GET for each location for now and hope to see other inputs on this matter.

**Comment by mackermann-ercot** [comment, marked as answer]
Hello @MandyZhang010 ,

cimview is correct.  We do not have a specification that would provide a means to query several values at once, however we do have a feature request for that exact behavior that has been accepted as a future enhancement (no ETA at this time, however).

Additionally, with over 160,000 records per 5 minutes, retrieving settlement point data for an entire range for a specific settlement point will be fairly slow.  Currently we are working on changing the internal query structure to provide exact matches on these query values in order to take advantage of additional indexes on the database.  

I would recommend that you query in smaller ranges.  Some users have had success searching for a specific settlement point (or bus, etc.)  for a smaller date range, then iterate through the ranges (i.e. a single settlement point for a single month, then the next month, etc.)

Regards,
-mackermann, ERCOT

**Reply by MandyZhang010** [reply]
Thanks for the reply! I will do a smaller batch download in this case and plz add that feature in your future development. 

Another question if i may. I saw on your other post that this API serves more of a historical archives purpose. If i want to query the spp/lmp following real-time publish interval (call every 15/5 minutes) would you suggest using this API? Or is there other method i can try?

**Reply by mackermann-ercot** [reply]
Hello again Mandy,

I apologize for the late response.  I did not get a notification for your follow-up question.  

Yes, you can use this API for retrieving this data at that interval, or download the files as well.

Regards,
-mackermann, ERCOT

**Reply by MandyZhang010** [reply]
Hi Mackermann - Thank you for confirming that. And I wanted to say I really appreciate ERCOT for setting up this discussion forum and the API project, it has been a great help for us who are working with historical data. And hope to see more data products be available via API in the future. Thanks again!
